src/bp-forums/groups.php

```diff
@@ -32,12 +32,235 @@ class BBP_Forums_Group_Extension extends BP_Group_Extension {
 		 * @since bbPress (r3552)
 		 */
 		public function __construct() {
+
+			// Redirect 404 page when user will input wrong URL from groups forum screen.
+			if ( ! $this->can_access() ) {
+				return;
+			}
+
 			$this->setup_variables();
 			$this->setup_actions();
 			$this->setup_filters();
 			$this->maybe_unset_forum_menu();
 		}
 
+		/**
+		 * Redirect 404 page when user will input wrong URL from groups forum screen.
+		 *
+		 * @since bbPress (r3552)
```

Comment:
@asaquzzamanbb why we have this @since bbPress (r3563) is this function forked from bbpress?

Reply:
Change `@since BuddyBoss 1.5.9`
https://github.com/buddyboss/buddyboss-platform/pull/2117/commits/da785ba8a0338d2bfbd7d4c829c6d554dfa69f8e, https://github.com/buddyboss/buddyboss-platform/pull/2117/commits/5890058f4ff3e80122785d2eb17b7dd25403a036, https://github.com/buddyboss/buddyboss-platform/pull/2117/commits/114240b580109fa63811374aa0a1f1d1f050cf6f